cmd/state-svc/internal/server/router.go

```diff
@@ -21,12 +19,7 @@ func (s *Server) setupRouting() {
 	})
 
 	s.httpServer.GET(QuitRoute, func(c echo.Context) error {
-		go func() {
-			err := s.Shutdown()
-			if err != nil {
-				logging.Errorf("Shutting down server failed: %v", err)
-			}
-		}()
+		s.Close()
```

Comment:
nit: Please consider keeping the term Shutdown. Generally, close implies cleanup of various allocations, and shutdown implies the same as well as some graceful management of things like client connections.